Editor/WorldFixedObjectProcessor.cs

```diff
@@ -86,16 +86,38 @@ private Transform CreateProxy()
             obj.transform.localRotation = Quaternion.identity;
             obj.transform.localScale = Vector3.one;
 
-            var constraint = obj.AddComponent<ParentConstraint>();
-            constraint.AddSource(new ConstraintSource()
+#if MA_VRCSDK3_AVATARS_3_7_0_OR_NEWER
+            var isVrcAvatar = avatarRoot.TryGetComponent(out VRC.SDKBase.VRC_AvatarDescriptor _);
+
+            if (isVrcAvatar)
+            {
+                var constraint = obj.AddComponent(
+                    System.Type.GetType("VRC.SDK3.Dynamics.Constraint.Components.VRCParentConstraint, VRC.SDK3.Dynamics.Constraint")
+                ) as VRC.Dynamics.ManagedTypes.VRCParentConstraintBase;
+                constraint.IsActive = true;
+                constraint.Locked = true;
+                constraint.AffectsPositionX = true;
+                constraint.AffectsPositionY = true;
+                constraint.AffectsPositionZ = true;
+                constraint.AffectsRotationX = true;
+                constraint.AffectsRotationY = true;
+                constraint.AffectsRotationZ = true;
+                constraint.FreezeToWorld = true;
+            }
+            else
+#endif
```

Comment:
I'd prefer here to extract this logic to a method and swap out the method as a whole, rather than having an else block spanning an `#endif` like this...

Reply:
IDK if the committed change is OK, it is hard to extract as a method because the case is "when VRCSDK with constraints is imported" (only can be filtered with `#if` preprocessor) and "we are working on a VRChat avatar" (only can be checked with `if (...) ... else` block)

Reply:
I'm thinking something like:

```
if (!TryGenerateVRChatConstraint(obj)) GenerateGenericConstraint(obj);

// ...
#if MA_VRCSDK3_AVATARS_3_7_0_OR_NEWER
bool TryGenerateVRChatConstraint(GameObject obj) { ... }
#else
bool TryGenerateVRChatConstraint(GameObject obj) { return false; }
#endif
```

Reply:
OK.